### ISSUE TITLE
Accuracy metric for preds at half precision is zero with pl=1.0.8

### ISSUE THREAD

**Issue by luzuku** [PyTorchLightning/pytorch-lightning#5013]
## 🐛 Bug

The accuracy metric is wrong if `preds` are given with half precision. See example. 

### To Reproduce

```python
import torch
from pytorch_lightning.metrics import Accuracy

acc = Accuracy(threshold=0.5)
target = torch.Tensor([1, 1, 0, 0])
preds = torch.Tensor([0.7, 0.4, 0.8, 0.4])

print(acc(preds, target))  -> 0.5
print(acc(preds.half(), target))  -> 0.0
```

### Expected behavior

The accuracy metric should not fail silently. Either an Error needs to be raised when preds are half precision or it should work correctly.

### Environment

 - PyTorch Version (e.g., 1.0): 1.7.0
 - OS (e.g., Linux): Linux
 - How you installed PyTorch (`conda`, `pip`, source): conda
 - Build command you used (if compiling from source):
 - Python version: 3.8
 - CUDA/cuDNN version: 10.2
 - GPU models and configuration: ...
 - Any other relevant information:

### Additional context

This might already be fixed in master. I filed the issue regardless because I don't have time to check.

**Comment by luzuku** [comment]
It is indeed fixed in `master` with the change from `preds.dtype == torch.float` to `preds.is_floating_point()`. It is also a duplicate of #4840. Sorry, my bad.

**Comment by tadejsv** [comment]
It is fixed in master but would be broken again in #4838, so thanks for catching this :)

**Comment by tadejsv** [comment]
@luzuku And one small note: if you create tensors with `torch.Tensor`, the created tensor will be a float tensor. Targets as floats will not be supported as inputs in classification functions anymore. If you want to preserve dtypes, create the tensor with `torch.tensor` (note that `tensor` is not capitalized)
